pages/glossary/actual-line-height.md

```diff
@@ -0,0 +1,23 @@
+---
+title: Actual Line Height
+key: actual-line-height
+unambiguous: true
+objective: true
+input_aspects:
+  - CSS styling
+  - DOM tree
+---
+
+The _actual line height_ of an element is obtained from the [computed value][] of its `line-height` property as follows:
+
+- if the value is a length, then the _actual line height_ is that length;
+- if the value is a number, then the _actual line height_ is that number multiplied by the [computed value][] of the `font-size` property of the element;
+- if the value is the keyword `normal`, then the _actual line height_ is 1.2 times the [computed value][] of the `font-size` property of the element.
+
+#### Background
+
+This definition is meant to mimic the [used value][] of the `line-height` property. However, in the `normal` case, this value depends on font specific metrics resulting in complex computations. Therefore, this definition uses 1.2 as a reasonable heuristic. [CSS specifications][line-height normal] recommend that the [used value][] is between 1.0 and 1.2 and major browsers are effectively using values close to 1.2.
+
+[computed value]: https://www.w3.org/TR/css-cascade-4/#computed 'CSS Cascading and Inheritance Level 4 (Working draft) - Computed Values'
+[line-height normal]: https://drafts.csswg.org/css2/#valdef-line-height-normal "CSS 2.2 (Editor's draft) - normal line-height"
+[used value]: https://www.w3.org/TR/css-cascade-4/#used 'CSS Cascading and Inheritance Level 4 (Working draft) - Used Values'
```

Comment:
This definition overloads the concept of "actual values" from CSS (https://www.w3.org/TR/css-cascade/#actual), which I'm not entirely comfortable with.

Reply:
Good point.
I wanted a name different from "used value" since it is not the same thing, but taking another existing name is as bad 🙈 

What about "effective line height"?

Reply:
I'd keep it simple and simply reference "used value" as that's the one we're really interested in.

Reply:
The problem is that it's not really the used value. When the `line-height` is `normal`, the used value can be many different things and we take 1.2 to keep it simple and because it is not too far from the used value in practice. But this is not really the used value (or not always).

Or you mean the rule could reference "used value" and let implementers decide what they want to do with that?

Reply:
> Or you mean the rule could reference "used value" and let implementers decide what they want to do with that?

Exactly. From the a rule specification perspective, we don't really care about the exact value of the given property, nor even the fact that it might differ between UAs. We just care about specifying something that is unambiguous.

To get the used `line-height` value, browser APIs such as `Node#clientHeight` can be used. Implementations without access to browser APIs can assume a used value when faced with `line-height: normal` or do something else; that's up to them.